llvm/lib/Transforms/Utils/Local.cpp

```diff
@@ -3423,6 +3424,17 @@ static void combineMetadata(Instruction *K, const Instruction *J,
                    MDNode::getMergedCallsiteMetadata(KCallSite, JCallSite));
   }
 
+  // Merge callee_type metadata.
+  // Handle separately to support cases where only one instruction has the
+  // metadata.
```

Comment:
I don't think this is correct? If one of the call does not have callee_type metadata, then we have to conservatively assume "all", so metadata should be dropped.

Reply:
The callee operand is generally not the same. If you adjust the SimplifyCFG test to use %fn and %fn2 you should get a call on `select` between the two.